Issue with ament_cmake_gen_version_h install path in ros2 build farm

Hello!

In ros2_control, we have recently added the version header generation using `ament_cmake_gen_version_h` package, however, from them, the build farm is failing showing the following error. Can someone help us in finding the issue in the changes (https://github.com/ros-controls/ros2_control/pull/1449/files)?. With the changes local builds and CI builds works without any issue.

Reference job : https://build.ros2.org/job/Rbin_uN64__joint_limits__ubuntu_noble_amd64__binary/61/
```
10:35:21 -- Installing: /tmp/binarydeb/ros-rolling-joint-limits-4.10.0/debian/ros-rolling-joint-limits/opt/ros/rolling/share/joint_limits/package.xml
10:35:21 CMake Error at cmake_install.cmake:145 (file):
10:35:21   file INSTALL cannot find
10:35:21   "/tmp/binarydeb/ros-rolling-joint-limits-4.10.0/.obj-x86_64-linux-gnu/ament_generate_version_header/joint_limits/joint_limits/version.h":
10:35:21   No such file or directory.
```

Thank you :)

--- a/ament_cmake_gen_version_h/cmake/ament_generate_version_header.cmake
+++ b/ament_cmake_gen_version_h/cmake/ament_generate_version_header.cmake
@@ -147,7 +147,7 @@ function(ament_generate_version_header target)
       "${SCRIPT_TEMPLATE_FILE}"
     COMMENT "Generating ${ARG_HEADER_PATH}")
 
-  add_custom_target("ament_generate_version_header__${target}"
+  add_custom_target("ament_generate_version_header__${target}" ALL
     DEPENDS "${GENERATED_HEADER_FILE}")
   add_dependencies("${target}" "ament_generate_version_header__${target}")
 

